packages/notification/src/features/utils.ts

```diff
@@ -0,0 +1,144 @@
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+
+import { getDefaultLanguage } from '@notification/i18n/utils'
+import {
+  CompositionSectionCode,
+  findCompositionSection,
+  findExtension,
+  getComposition,
+  getResourceFromBundleById,
+  getStatusFromTask,
+  getTaskFromSavedBundle,
+  Location,
+  Patient,
+  ReadyForReviewRecord,
+  RegisteredRecord,
+  RelatedPerson,
+  resourceIdentifierToUUID,
+  urlReferenceToUUID
+} from '@opencrvs/commons/types'
+import { badRequest as boomBadRequest } from '@hapi/boom'
+
+export function getContactPhoneNo(
+  record: ReadyForReviewRecord | RegisteredRecord
+) {
+  const task = getTaskFromSavedBundle(record)
+  const phoneNumberExtension = findExtension(
+    'http://opencrvs.org/specs/extension/contact-person-phone-number',
+    task.extension
+  )
+  return phoneNumberExtension?.valueString
+}
+
+export function getContactEmail(
+  record: ReadyForReviewRecord | RegisteredRecord
+) {
+  const task = getTaskFromSavedBundle(record)
+  const emailExtension = findExtension(
+    'http://opencrvs.org/specs/extension/contact-person-email',
+    task.extension
+  )
+  return emailExtension?.valueString
+}
+
+function error(
+  record: ReadyForReviewRecord | RegisteredRecord,
+  message: string
+): never {
+  const task = getTaskFromSavedBundle(record)
+  const taskStatus = getStatusFromTask(task)
+  throw boomBadRequest(`${message} in ${taskStatus} record`)
+}
+
+export function getOfficeName(record: ReadyForReviewRecord | RegisteredRecord) {
+  const task = getTaskFromSavedBundle(record)
+  const officeExtension = findExtension(
+    'http://opencrvs.org/specs/extension/regLastOffice',
+    task.extension
+  )
+  if (!officeExtension?.valueString) {
+    error(record, 'Office extension not found')
+  }
+  return officeExtension.valueString
+}
+
+export function getInformantName(
+  record: ReadyForReviewRecord | RegisteredRecord
+) {
+  const composition = getComposition(record)
+  const informantSection = findCompositionSection(
+    'informant-details',
+    composition
+  )
+  if (!informantSection) {
+    error(record, 'informant section not found')
+  }
+  const informantRelation = getResourceFromBundleById<RelatedPerson>(
+    record,
+    urlReferenceToUUID(informantSection.entry[0].reference)
+  )
+  const informant = getResourceFromBundleById<Patient>(
+    record,
+    urlReferenceToUUID(informantRelation.patient.reference)
+  )
+  const name = informant.name.find(({ use }) => use === 'en')
+  if (!name) {
+    error(record, 'name not found in informant patient resource')
+  }
+  return [name.given?.join(' '), name.family.join(' ')].join(' ').trim()
+}
+
+export function getPersonName(
+  record: ReadyForReviewRecord | RegisteredRecord,
+  compositionCode: Extract<
+    CompositionSectionCode,
+    'deceased-details' | 'child-details'
+  >
+) {
```

Comment:
We can improve on this by taking a person parameter `'child' | 'deceased'` and then mapping them to the appropriate `CompositionSectionCode` inside the function, that should make the function's interface much more readable